examples/paho-mqtt/Makefile

```
@@ -0,0 +1,59 @@
+APPLICATION = paho-mqtt-example
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+# Comment this out to disable code in RIOT that does safety checking
+# which is not needed in a production environment but helps in the
+# development process:
+DEVELHELP ?= 1
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+WIFI_SSID ?= "Your_WiFi_name"
+WIFI_PASS ?= "Your_secure_password"
+
+USEMODULE += shell
+USEMODULE += shell_commands
+USEMODULE += ps
+USEMODULE += netdev_default
+USEPKG += paho-mqtt
+
+# paho-mqtt depends on TCP support, choose the stack you want
+LWIP_IPV4 ?= 0
+
+ifneq (0, $(LWIP_IPV4))
+  USEMODULE += ipv4_addr
+  USEMODULE += lwip_arp
+  USEMODULE += lwip_ipv4
+  USEMODULE += lwip_dhcp_auto
+  USEMODULE += lwip_sock_udp
+  CFLAGS += -DETHARP_SUPPORT_STATIC_ENTRIES=1
+  LWIP_IPV6 ?= 0
+else
+  LWIP_IPV6 ?= 1
+endif
+
+ifneq (0, $(LWIP_IPV6))
+  USEMODULE += ipv6_addr
+  USEMODULE += lwip_ipv6_autoconfig
+endif
+
+USEMODULE += lwip_netdev
+USEMODULE += lwip lwip_sock_ip
+USEMODULE += lwip_tcp lwip_sock_tcp
+USEMODULE += lwip_sock_async
+
+USEMODULE += sock_async_event
+####
+
+include $(RIOTBASE)/Makefile.include
+
+ifneq (,$(filter arch_esp,$(FEATURES_USED)))
+  CFLAGS += -DESP_WIFI_SSID=\"$(WIFI_SSID)\"
+  CFLAGS += -DESP_WIFI_PASS=\"$(WIFI_PASS)\"
+endif
```

Sorry, I forgot to mention that the ESP8266 requires an increase in stack size to 2048 bytes to get it running without crashes. The default stack size for ESP8266 is only 1024 bytes. I had always defined it as a command line option and therefore forgot to request the following change.
```suggestion
  CFLAGS += -DMQTT_THREAD_STACKSIZE=2048
endif
```

---

This will change the stack size even for the ESP32... do we want this? I have to confess that I'm not always testing on ESP32 but when I did, the stack size seemed OK.

---

Maybe set the `MQTT_THREAD_STACKSIZE` to [`THREAD_STACKSIZE_LARGE`](https://doc.riot-os.org/group__core__thread.html#gafab9ad4e0af6c50af1ff9e1fb5ebfd1f) (double the default stack size) as a default? I can't imagine the ESP32 and ESP8266 are the only ones having problems and for `native` the stack size is basically "free".

---

The standard stack size for ESP32 is already 2048 bytes. So it does not matter. I just don't want to have a `if ...` in the `Makefile` just for this setting. In fact, a stack size of 2048 bytes fits both ESPs, regardless of what the default stack sizes will be in the future.

---

That's ok for me, what do you think @gschorcht ?

---

Oups... it seems we both commented at the same time :$

---

> Maybe set the `MQTT_THREAD_STACKSIZE` to [`THREAD_STACKSIZE_LARGE`](https://doc.riot-os.org/group__core__thread.html#gafab9ad4e0af6c50af1ff9e1fb5ebfd1f) (double the default stack size) as a default? I can't imagine the ESP32 and ESP8266 are the only ones having problems and for `native` the stack size is basically "free".

For the ESP32 it is not necessary to double the stack size with 'THREAD_STACKSIZE_LARGE'.

---

> For the ESP32 it is not necessary to double the stack size with 'THREAD_STACKSIZE_LARGE'.

But if `THREAD_STACKSIZE_LARGE` avoids potential problems on other platforms, I'm fine with it. The RAM on ESP32 is usually not a problem.

---

> The RAM on ESP32 is usually not a problem.

Exactly, and if the user comes into RAM problems, trying to decrease stack sizes via configuration might be the first thing they try ;-).